content/blog/2019-05-14-nasapower.md

```diff
@@ -0,0 +1,625 @@
+data you can always have a look at the list of available parameters
+using `?parameters`. That shows that T2M\_MIN and T2M\_MAX
+are the minimum and maximum temperatures at 2 meters. We will use those
+to construct the query.
+
+``` r
+library(nasapower)
+
+dv <- get_power(community = "AG",
+                lonlat = c(-117.2180, 36.7266),
+                dates = c("1983-01-01", "2018-12-31"),
+                temporal_average = "DAILY",
+                pars = c("T2M_MIN", "T2M_MAX"))
+
+dv
+```
+
+    ## NASA/POWER SRB/FLASHFlux/MERRA2/GEOS 5.12.4 (FP-IT) 0.5 x 0.5 Degree Daily Averaged Data  
+    ##  Dates (month/day/year): 01/01/1983 through 12/31/2018  
+    ##  Location: Latitude  36.7266   Longitude -117.218  
+    ##  Elevation from MERRA-2: Average for 1/2x1/2 degree lat/lon region = 1192.53 meters   Site = na  
+    ##  Climate zone: na (reference Briggs et al: http://www.energycodes.gov)  
+    ##  Value for missing model data cannot be computed or out of model availability range: -99  
+    ##  
+    ##  Parameters: 
+    ##  T2M_MIN MERRA2 1/2x1/2 Minimum Temperature at 2 Meters (C) ;
+    ##  T2M_MAX MERRA2 1/2x1/2 Maximum Temperature at 2 Meters (C)  
+    ##  
+    ## # A tibble: 13,149 x 9
+    ##      LON   LAT  YEAR    MM    DD   DOY YYYYMMDD   T2M_MIN T2M_MAX
+    ##    <dbl> <dbl> <dbl> <int> <int> <int> <date>       <dbl>   <dbl>
+    ##  1 -117.  36.7  1983     1     1     1 1983-01-01   -4.73    7.24
+    ##  2 -117.  36.7  1983     1     2     2 1983-01-02   -3.33    7.72
+    ##  3 -117.  36.7  1983     1     3     3 1983-01-03    0.05   11.8 
+    ##  4 -117.  36.7  1983     1     4     4 1983-01-04    1.27   14.9 
+    ##  5 -117.  36.7  1983     1     5     5 1983-01-05    2.55   15.9 
+    ##  6 -117.  36.7  1983     1     6     6 1983-01-06    2.63   17.1 
+    ##  7 -117.  36.7  1983     1     7     7 1983-01-07    3.2    17.2 
+    ##  8 -117.  36.7  1983     1     8     8 1983-01-08    1.96   18.3 
+    ##  9 -117.  36.7  1983     1     9     9 1983-01-09    0.7    14.0 
+    ## 10 -117.  36.7  1983     1    10    10 1983-01-10    1.3    17.6 
+    ## # … with 13,139 more rows
+
+In the metadata header you can see information about where the data
+comes from and what dates have been queried and returned as well as the
+elevation data.
+
+The `tibble` contains a few columns not in the original data, but that
+can make it easier to work within R. The original data only include YEAR
+and DOY. Looking at the data returned, there are LON = the queried
+longitude as double; LAT = the queried latitude as double; YEAR = the
+queried year as double; MM = the queried month as double, DD = the
+queried day as double, DOY = the day of year or Julian date as integer,
+YYYYMMDD = the full date as a date object and the requested parameters
+T2M\_MIN and T2M\_MAX each as double.
+
+#### Visualising the Data
+
+To visualise these data I will use ggplot2, but first I need to gather the data
```

Comment:
Again, I like how you explain what you are about to do, and then after some output, you give an interpretation. So good for readers